plugin/storage/cassandra/factory.go

```diff
@@ -78,6 +80,56 @@ func NewFactory() *Factory {
 	}
 }
 
+// NewFactoryWithConfig initializes factory with Config.
+func NewFactoryWithConfig(
+	cfg config.Configuration,
+	metricsFactory metrics.Factory,
+	logger *zap.Logger,
+) (*Factory, error) {
+	archive := make(map[string]*namespaceConfig)
+	serverURL, err := getServers(cfg.Servers)
+	if err != nil {
+		return nil, err
+	}
+	archive[archiveStorageConfig] = &namespaceConfig{
+		Configuration: cfg,
+		servers:       serverURL,
+		namespace:     archiveStorageConfig,
+		Enabled:       true,
+	}
+
+	f := NewFactory()
+	f.InitFromOptions(&Options{
+		Primary: namespaceConfig{
+			Configuration: cfg,
+			servers:       serverURL,
+			namespace:     primaryStorageConfig,
+			Enabled:       true,
+		},
+		others:                 archive,
+		SpanStoreWriteCacheTTL: time.Hour * 12,
+	})
+	err = f.Initialize(metricsFactory, logger)
+	if err != nil {
+		return nil, err
+	}
+	return f, nil
+}
+
+func getServers(servers []string) (string, error) {
```

Comment:
Not following why we need to go from structured array back to a string

Reply:
We are converting an array to a string because namespaceConfig.servers want a string. Which on further processing in f.Options.GetPrimary() or f.Options.Get() gets converted back to []string.

if we don't want this conversion. we have to change it in these functions.

Reply:
ok, but we don't need to be calling InitFromOptions function, which is designed to transfer information from CLI flags into Config. You already have config here, just assign it to the factory, and we wouldn't need these shenanigans.